tests/test_local_executor.py

```diff
@@ -401,6 +427,68 @@ def test_finalize_failed_oomkilled(docker_cleanup, job, tmp_work_dir):
         == "Ran out of memory (limit for this job was 0.01GB)"
     )
 
+    assert workspace_log_file_exists(job)
+
+
+@pytest.mark.needs_docker
+def test_pending_job_cancelled_not_finalized(docker_cleanup, job, tmp_work_dir):
```

Comment:
```suggestion
def test_pending_job_terminated_not_finalized(docker_cleanup, job, tmp_work_dir):
```

Cancelled is a run.py concept, not an executor concept, which only has `terminated()`

Reply:
ah, yes, you're right, I wanted to differentiate from https://github.com/opensafely-core/job-runner/blob/master/tests/test_local_executor.py#L383 but now I look again at that, that's `kill` not `terminate`